File: 169/Solution_169.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> hashmap;
        
        // Populate the hashmap with element frequencies
        for (int i = 0; i < nums.size(); i++) {
            hashmap[nums[i]]++;
        }

        // Find the majority element
        int majority = nums.size() / 2;
        int result = 0;
        for (const auto& e : hashmap) {
            int element = e.first;
            int count = e.second;
            if (count > majority) {
                result = element;
                break;
            }
        }

        return result;
    }
};


int main(){
    Solution solution;
    vector<int> nums = {3, 2, 3};
    cout << "Majority Element: " << solution.majorityElement(nums) << endl;
    return 0;
}

File: 169/Solution_169.java
//方法二：排序

import java.util.Arrays;

public class Solution_169 {
    
    public static void main(String[] args){
        Solution solution = new Solution();
        int[] nums = {3, 2, 3};
        System.out.println("多数元素是: " + solution.majorityElement(nums)); // 输出: 3

        int[] nums2 = {2, 2, 1, 1, 1, 2, 2};
        System.out.println("多数元素是: " + solution.majorityElement(nums2)); // 输出: 2
    }
}

class Solution {

    public int majorityElement( int[] nums ){
        Arrays.sort(nums);
        return nums[nums.length / 2];
    }
}
